### PR TITLE
Translate Plugin - Target Type implementation

### PR DESCRIPTION
**Pull request by vishalboin**
### Description
Implemented target_type option in translate processor. Added necessary test cases.
 
### Issues Resolved
https://github.com/opensearch-project/data-prepper/issues/1914
 
### Check List
- [x] New functionality includes testing.
- [x] New functionality has been documented.
- [ ] New functionality has javadoc added
- [x] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-plugins/translate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/translate/TranslateProcessorConfig.java [hunk 1]
```diff
@@ -85,4 +95,30 @@ public boolean isPatternPresent(){
         return regexParameterConfiguration == null || regexParameterConfiguration.getPatterns() != null;
     }
 
+    @AssertTrue(message = "The mapped values does not match the target type provided")
```
Comment (kkondaka) [inline data-prepper-plugins/translate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/translate/TranslateProcessorConfig.java:98]: Nit: "The mapped values do not match the target type provided"

File: data-prepper-plugins/translate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/translate/TranslateProcessorConfig.java [hunk 2]
```diff
@@ -85,4 +95,30 @@ public boolean isPatternPresent(){
...
+    public boolean isMapTypeValid() {
+        return map.keySet().stream().allMatch(key -> checkTargetValueType(map.get(key)));
+    }
+
+    @AssertTrue(message = "The pattern values does not match the target type provided")
```
Comment (kkondaka) [inline data-prepper-plugins/translate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/translate/TranslateProcessorConfig.java:103]: Nit: same as above "does not" -> "do not"

File: data-prepper-plugins/translate-processor/build.gradle [hunk 3]
```diff
@@ -16,6 +16,7 @@ dependencies {
     testImplementation project(':data-prepper-plugins:log-generator-source')
     testImplementation project(':data-prepper-test-common')
     implementation 'org.apache.commons:commons-lang3:3.12.0'
+    testImplementation project(path: ':data-prepper-plugins:mutate-event-processors')
```
Comment (graytaylor0) [inline data-prepper-plugins/translate-processor/build.gradle:19]: Why is this `testImplementation`?

File: data-prepper-plugins/translate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/translate/TranslateProcessorConfig.java [hunk 4]
```diff
@@ -10,14 +10,19 @@
...
 
 import java.util.List;
 import java.util.Map;
+import java.util.Objects;
 import java.util.stream.Stream;
 
 
 public class TranslateProcessorConfig {
 
+    final TargetType TARGET_TYPE = TargetType.STRING;
```
Comment (graytaylor0) [inline data-prepper-plugins/translate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/translate/TranslateProcessorConfig.java:24]: Nit: Could just put this inline

```
private TargetType targetType = TargetType.STRING
```

File: data-prepper-plugins/translate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/translate/TranslateProcessor.java [hunk 5]
```diff
@@ -40,15 +42,17 @@ public class TranslateProcessor extends AbstractProcessor<Record<Event>, Record<
     private static final Logger LOG = LoggerFactory.getLogger(TranslateProcessor.class);
     private final ExpressionEvaluator expressionEvaluator;
     private final TranslateProcessorConfig translateProcessorConfig;
-    private final LinkedHashMap<Range<Float>, String> rangeMappings;
-    private final Map<String, String> individualMappings;
-    private final Map<Pattern, String> compiledPatterns;
+    private final LinkedHashMap<Range<Float>, Object> rangeMappings;
+    private final Map<String, Object> individualMappings;
+    private final Map<Pattern, Object> compiledPatterns;
+    private final TypeConverter converter;
```
Comment (graytaylor0) [inline data-prepper-plugins/translate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/translate/TranslateProcessor.java:48]: Another option to not take a dependency on all mutate-events processors would be for us to just move the convert processor out to its own module

File: data-prepper-plugins/translate-processor/build.gradle (hunk 3 above)
Comment (vishalboin) [reply data-prepper-plugins/translate-processor/build.gradle:19]: Wrong dependency. I with push changes with `implementation project(path: ':data-prepper-plugins:mutate-event-processors')`.

File: data-prepper-plugins/translate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/translate/TranslateProcessor.java (hunk 5 above)
Comment (vishalboin) [reply data-prepper-plugins/translate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/translate/TranslateProcessor.java:48]: I will do it in the next PR

Comment (kkondaka) [reply data-prepper-plugins/translate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/translate/TranslateProcessor.java:48]: I think this translate processor needs to be moved into mutate-event-processors

**Review by graytaylor0 — APPROVED** [review]

**Review by kkondaka — APPROVED** [review]

**Pull request merged**